src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestCommentCause.java

```diff
@@ -1,18 +1,22 @@
 package io.jenkins.plugins.gitlabbranchsource;
 
 import hudson.model.Cause;
+import io.jenkins.plugins.gitlabbranchsource.Cause.GitLabMergeRequestNoteData;
+import org.gitlab4j.api.webhook.NoteEvent;
 
 public final class GitLabMergeRequestCommentCause extends Cause {
 
     private final String commentUrl;
+    private final GitLabMergeRequestNoteData gitLabMergeRequestNoteData;
 
     /**
      * Constructor.
      *
      * @param commentUrl the URL for the GitLab comment
      */
-    public GitLabMergeRequestCommentCause(String commentUrl) {
+    public GitLabMergeRequestCommentCause(String commentUrl, NoteEvent noteEvent) {
```

Comment:
I prefer to add a new constructor instead of changing the existing one. Because you cannot sure there are no others who might need that.

Reply:
Good point, sir